tests/_pure_prompt_nixdevshell.test.fish

```diff
@@ -22,14 +23,31 @@ before_each
     set --universal pure_enable_nixdevshell true
     set --erase IN_NIX_SHELL
 
-    _pure_prompt_nixdevshell
+    # when executing the tests in nixos builds, /nix/store paths are in the $PATH variable
+    begin
```

Comment:
Woo, thanks! The `begin… set --local … end` sound cool I shall use it more often

Reply:
I've just googled how to set a variable for child scopes and automatically drop it afterwards. otherwise I would break everything by overwriting path.